server/events.go

```diff
@@ -417,6 +417,9 @@ func (s *Server) sendInternalMsg(sub, rply string, si *ServerInfo, msg interface
 
 // Locked version of checking if events system running. Also checks server.
 func (s *Server) eventsRunning() bool {
+	if s == nil {
```

Comment:
Weird that we have to check for `s == nil` here.. I would instead have fixed the call stack to find out when this gets invoked with a nil server.

Reply:
Was in a test. So nothing real per se but added it since it crashed when I did not check that the server was nil in the test and called Shutdown on it.

Reply:
I would fix the test to be honest. This check is happening all the time (for no good reason) but more importantly, as a reader, it says that it is "normal" in some situations to have s being nil. I am guilty of doing that too sometimes, but I really try hard not to check against nil if the object is not meant to be nil, ever.

Reply:
I did fix test but don't mind the protection for nil receivers tbh.